### PR TITLE
ascli: Allow searching by categories

### PR DESCRIPTION
**Pull request by aleixpol**
Allows the command line interface to be used to list available components for certain categories.

### CODE REVIEW

**Comment by aleixpol** [comment]
I set it up for debugging purposes, I figured it might be useful to others, as it's part of the public API anyway.

File: tools/ascli-actions-mdata.c [hunk 1]
```diff
@@ -252,6 +252,50 @@ ascli_what_provides (const gchar *cachepath, const gchar *kind_str, const gchar
...
+        return 2;
+    }
+
+    dpool = ascli_data_pool_new_and_open (cachepath, no_cache, &error);
+    if (error != NULL) {
+        g_printerr ("%s\n", error->message);
+        return 1;
+    }
+
+    g_auto(GStrv) cats = g_strsplit(categories, ",", 0);
```
Comment (ximion) [inline tools/ascli-actions-mdata.c:278]: No declarations after statements please, we (still) need to be compatible with older C versions/compilers.

File: tools/ascli-actions-mdata.c [hunk 2]
```diff
@@ -252,6 +252,50 @@ ascli_what_provides (const gchar *cachepath, const gchar *kind_str, const gchar
...
+
+    result = as_pool_get_components_by_categories (dpool, cats);
+    if (result->len == 0) {
+        /* TRANSLATORS: We failed to find any component in the database, likely due to an error */
+        ascli_print_stderr (_("Unable to find components matching %s!"), categories);
```
Comment (ximion) [inline tools/ascli-actions-mdata.c:283]: nit: Maybe just call it `Unable to find components in categories "%s"!` to make this more clear?

**Review by ximion — CHANGES_REQUESTED** [review]

File: tools/appstreamcli.c [hunk 3]
```diff
@@ -1252,6 +1282,12 @@ as_client_run (char **argv, int argc)
 			/* TRANSLATORS: `appstreamcli what-provides` command description. */
 			_("Get components which provide the given item. Needs an item type (e.g. lib, bin, python3, …) and item value as parameter."),
 			as_client_run_what_provides);
+	ascli_add_cmd (commands,
+			0, "by-categories", NULL, "NAMES",
```
[the remark this answers is not in the document]
Comment (aleixpol) [reply tools/appstreamcli.c:1286]: Sure, I don't mind either way. In fairness, I'd forgotten about this PR already.

File: tools/ascli-actions-mdata.c (hunk 1 above)
Comment (aleixpol) [reply tools/ascli-actions-mdata.c:278]: I don't think I've ever seen a compiler that _required_ it and I've been doing C for about 20 years now.

**Pull request closed without merging**